src/react/components/pages/editorPage/canvas.tsx

```diff
@@ -0,0 +1,329 @@
+     * @param {string} id
+     * @param {RegionData} regionData
+     * @param {TagsDescriptor} tagsDescriptor
+     * @returns {void}
+     */
+    public addRectRegion: (id: string, regionData: RegionData, tagsDescriptor: TagsDescriptor) => void;
+
+    /**
+     * @name deleteAllRegions
+     * @description deletes all regions from the canvas
+     * @returns {void}
+     */
+    public deleteAllRegions: () => void;
+
+    /**
+     * @name deleteRegionById
+     * @description deletes region with given id from the canvas
+     * @param {string} id
+     * @returns {void}
+     */
+    public deleteRegionById: (id: string) => void;
+
+    /**
+     * @name freeze
+     * @description freezes all regions the canvas to prevent accidental edits (used for hotkey control)
+     * @param {string} nuance
+     * @returns {void}
+     */
+    public freeze: (nuance: string) => void;
+
+    /**
+     * @name getSelectedRegionsBounds
+     * @description get the boinding box info for the current region
+     * @returns {Object}
+     */
+    public getSelectedRegionsBounds: () => {id: string, x: number, y: number, width: number, height: number};
+
+    /**
+     * @name redrawAllRegions
+     * @description clears all regions from the canvas and redraws them
+     * @returns {void}
+     */
+    public redrawAllRegions: () => void;
+
+    /**
+     * @name resize
+     * @description resize editor canvas
+     * @param {number} width
+     * @param {number} height
+     * @returns {void}
+     */
+    public resize: (width: number, height: number) => void;
+
+    /**
+     * @name selectRegionById
+     * @description selects region with given id on the canvas
+     * @param {string} id
+     * @returns {void}
+     */
+    public selectRegionById: (id: string) => void;
+
+    /**
+     * @name toggleFreezeMode
+     * @description toggles between freezing/unfreezing all regions on the canvas
+     * @returns {void}
+     */
+    public toggleFreezeMode: () => void;
+
+    /**
+     * @name unfreeze
+     * @description unfreezes all regions on the canvas
+     * @returns {void}
+     */
+    public unfreeze: () => void;
+
+    /**
+     * @name updateTagsById
+     * @description uppdates region with given id with provided tags descriptor
+     * @param {string} id
+     * @param {TagsDescriptor} tagsDescriptor
+     * @returns {void}
+     */
+    public updateTagsById: (id: string, tagsDescriptor: TagsDescriptor) => void;
+
+    /**
+     * @name updateTagsForSelectedRegions
+     * @description uppdates tags for currently selected region with provided tags descriptor
+     * @param {TagsDescriptor} tagsDescriptor
+     * @returns {void}
+     */
+    public updateTagsForSelectedRegions: (tagsDescriptor: TagsDescriptor) => void;
+    private editor: Editor;
+
+    constructor(props, context) {
+        super(props, context);
+
+        this.state = {
+            loaded: false,
+        };
+    }
+
+    public componentDidMount() {
```

Comment:
Seems like this could use a bit of cleanup. Maybe add some functions that are called by this one (`getTags`, `onRegionMove`, `onRegionDelete`, etc.) This function is just big, which makes it a little more challenging to read